### PR TITLE
Added Auth for upload API

### PR DESCRIPTION
**Pull request by IlyaMoskva**
#83 Auth for SliceUpload

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#86](https://github.com/artipie/maven-adapter/issues/86) is assigned to @victornoel[/z](https://www.0crat.com/u/victornoel), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/16b69953-9759-40e6-b437-2ee490aeeb2d); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/main/java/com/artipie/maven/http/MavenSlice.java [hunk 1]
```diff
@@ -78,15 +78,19 @@ private MavenSlice(final Storage storage, final Permissions perms, final Identit
...
+                        new Permission.ByName("download", perms),
+                        users
+                    )
                 ),
                 new SliceRoute.Path(
                     new RtRule.ByMethod(RqMethod.PUT),
-                    new SliceUpload(storage)
+                    new SliceAuth(
+                        new SliceUpload(storage),
+                        new Permission.ByName("upload", perms),
```
Comment (victornoel) [inline src/main/java/com/artipie/maven/http/MavenSlice.java:91]: @IlyaMoskva can you add a test for this?

**Comment by victornoel** [comment]
@IlyaMoskva ping

**Comment by victornoel** [comment]
@IlyaMoskva can you take the comment into account?

**Comment by 0crat** [comment]
@victornoel[/z](https://www.0crat.com/u/victornoel) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by g4s8** [comment]
@victornoel @IlyaMoskva  we needs this changes right, not so we can implement the test later

**Comment by victornoel** [comment]
@g4s8 I wasn't really finished and was waiting for the test, or at least a todo. Next time can you give me some time to react to your requirement?

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @victornoel[/z](https://www.0crat.com/u/victornoel), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@victornoel please make sure you will find at least 3 issues during next CR, thanks!

@IlyaMoskva please make sure that all problems found by the code reviewer (@victornoel) were addressed by the pull request author (you, @IlyaMoskva), which means that they were either fixed as the reviewer wanted or the reviewer was convinced that they should not be fixed

**Comment by victornoel** [comment]
@sereshqua yes, if ARC let me do it :P

**Comment by sereshqua** [comment]
@IlyaMoskva please confirm and I will give verdict

**Comment by sereshqua** [comment]
@IlyaMoskva ping

**Comment by IlyaMoskva** [comment]
@sereshqua Ok, got it. Tnx.

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Pull request merged**